kratos/utilities/variable_utils.h

```diff
@@ -800,24 +800,29 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     {
         KRATOS_TRY
 
-        if(rNodes.size() != 0) {
-            // First we do a check
-            CheckVariableExists(rVar, rNodes);
+        if (rNodes.size() != 0) {
+            // checking the first node to avoid error being thrown in parallel region
+            KRATOS_ERROR_IF_NOT(rNodes.begin()->HasDofFor(rVar)) << "Trying to fix/free dof of variable " << rVar.Name() << " but this dof does not exist in node #" << rNodes.begin()->Id() << "!" << std::endl;
 
-            // I assume that all the nodes sharing the same varibles list from modelpart
-            rNodes.begin()->pGetVariablesList()->AddDof(&rVar);
+#ifdef KRATOS_DEBUG
+            for (const auto& r_node : rNodes) {
+                KRATOS_ERROR_IF_NOT(r_node.HasDofFor(rVar)) << "Trying to fix/free dof of variable " << rVar.Name() << " but this dof does not exist in node #" << r_node.Id() << "!" << std::endl;
+            }
+#endif
+
+            CheckVariableExists(rVar, rNodes);
 
-            if(IsFixed == true) {
+            if (IsFixed) {
                 #pragma omp parallel for
                 for (int k = 0; k< static_cast<int> (rNodes.size()); ++k) {
                     NodesContainerType::iterator it_node = rNodes.begin() + k;
-                    it_node->pAddDof(rVar)->FixDof();
+                    it_node->pGetDof(rVar)->FixDof();
```

Comment:
@KratosMultiphysics/technical-committee :
We would go for directly fixing without the second check, since this would imply a second search and also it can hide problems in the usage like we detected here in the embedded solver